src/liboslexec/batched_analysis.cpp

```diff
@@ -2008,7 +2002,13 @@ struct Analyzer {
     {
         bool previously_was_uniform = symbol_to_be_varying->is_uniform();
         if (previously_was_uniform | force) {
-            symbol_to_be_varying->make_varying();
+            // Interactive params are never varying, following getLLVMSymbolBase()
```

Comment:
While it is technically correct that interactive params are always uniform, we shouldn't have to force it.
The need for this change show's that there is bug elsewhere.



Previously there was just Symbol::lockgeom() and that changed to Symbol::interpolated() and Symbol::interactive() and the lockgeom covered both:

    bool Symbol::lockgeom() const
    {
        // We can lock a value to a constant (at all places on all pieces of
        // geometry) if it is neither interpolated nor interactively modified.
        return !m_interpolated && !m_interactive;
    }


However not all code that used lockgeom got updated, in particular the method:    


    void Analyzer::make_interpolated_parameters_varying()
    {
        // Mark all interpolated parameters as varying,
        // As we expect interpolated data,  get_userdata will
        // only provide varying values, so we must mark
        // our symbols appropriately!
        FOREACH_PARAM(Symbol & s, inst())
        {
            if (s.everread() && !s.lockgeom() && !s.typespec().is_closure()) {
                recursively_mark_varying(&s);
            }
        }
    }


Could you try changing it from !lockgeom() to interactive() there:

    void Analyzer::make_interpolated_parameters_varying()
    {
        // Mark all interpolated parameters as varying,
        // As we expect interpolated data,  get_userdata will
        // only provide varying values, so we must mark
        // our symbols appropriately!
        FOREACH_PARAM(Symbol & s, inst())
        {
            if (s.everread() && s.interpolated() && !s.typespec().is_closure()) {
                recursively_mark_varying(&s);
            }
        }
    }

  
Then replace your force_uniform change with an
    `OSL_ASSERT(!symbol_to_be_varying->interactive());`